### ISSUE TITLE
tooling for on device replays (without hardware)

### ISSUE THREAD

**Issue by darvin** [commaai/openpilot-tools#5]
is it conceivable to make on device video/can messages replays without hardware? of course, not streaming, but say reading from the files, on the loop.

reasoning: would greatly simplify work on Openpilot UI, without need to acquire development hardware

**Comment by rbiasini** [comment]
Yes, I would support this change. It would just need some minimal changes to `ui.c`in openpilot to optionally subscribe to a frame coming from a pc.

**Comment by interose** [comment]
Hi, are there any news regarding this topic?
Is it possible to use the "Replay driving data" tool with any camera in order to get a basic idea?
